scripts/xtensa-build-zephyr.sh

```diff
@@ -134,11 +137,11 @@ build()
 				PLAT_CONFIG='intel_adsp_cavs25'
 				XTENSA_CORE="cavs2x_LX6HiFi3_2017_8"
 				XTENSA_TOOLS_VERSION="RG-2017.8-linux"
-				RIMAGE_KEY=modules/audio/sof/keys/otc_private_key_3k.pem
+				rimage_key=modules/audio/sof/keys/otc_private_key_3k.pem
 				;;
 			imx8)
 				PLAT_CONFIG='nxp_adsp_imx8'
-				RIMAGE_KEY='' # no key needed for imx8
+				rimage_key="no signing for $platform"
```

Comment:
does this mean, that now rimage command on imx8 will actually be executed with a `-k "no signing for imx8"` parameter? I understand that it should be ignored by rimage, but I'm not sure it's a good thing for correctness. Wouldn't the best option be to omit `-k` on such platforms altogether?

Reply:
What would be the command line for NOT signing with this change, sorry not following.

Reply:
Now, in `rimage` `-k` is a mandatory option - see https://github.com/thesofproject/rimage/blob/main/src/rimage.c#L109
That's why for imx, I used an empty string as key (`RIMAGE_KEY=''`).
I agree with @lyakh it's better to omit `-k`, to make it optional, but that would mean we should modify `rimage`.

Reply:
Ack - lets make -k optional

Reply:
I changed `-k ''` to `-k 'no signing for imx8`' because the former wasted a fair amount of my time; the latter would have not. Among others, I could have `git grep`ped the latter instantly. Also makes the logs much more obvious, no risk of assuming rimage is falling back on a default key or something like that.

> Ack - lets make -k optional

@lgirdwood this is artificially blocking this PR = the fix of pretty serious bug in this script (using the wrong key) on a nice to have (no fake rimage argument on the command line). Plus all Intel platforms use a key.

Reply:
Lets just keep it simple and fix rimage. -k can be optional where the -k argument is mandatory. Then ```image.key_name == NULL``` means the signing part can be easily skipped in rimage by not calling the signing function (i.e. the output structure is the same as before but with 0s where the signature should be)

Reply:
Sounds great but I'm not volunteering sorry; already spending way too much time fixing others' problems. I rebased and removed the empty `-k` change, now this PR is smaller and a pure and totally unrelated fix. As a bonus of being a smaller change it will not conflict with the future revert of #4719 when XCC builds are re-enabled.